frame/staking/src/pallet/impls.rs

```diff
@@ -737,18 +737,12 @@ impl<T: Config> Pallet<T> {
 				None => break,
 			};
 
-			if let Some(Nominations { submitted_in, mut targets, suppressed: _ }) =
-				<Nominators<T>>::get(&voter)
-			{
-				// if this voter is a nominator:
-				targets.retain(|stash| {
-					slashing_spans
-						.get(stash)
-						.map_or(true, |spans| submitted_in >= spans.last_nonzero_slash())
-				});
-				if !targets.len().is_zero() {
+			if let Some(Nominations { targets, .. }) = <Nominators<T>>::get(&voter) {
+				if !targets.is_empty() {
 					all_voters.push((voter.clone(), weight_of(&voter), targets));
 					nominators_taken.saturating_inc();
+				} else {
+					// Technically should never happen, but not much we can do about it. 
```

Comment:
Should there be a debug statement at least? Otherwise why bother with an `else` branch?

Reply:
Just for readability.

Reply:
I'd prefer a simple comment someplace to if/else that does nothing. But I would also describe this as personal preference. Empty if/else blocks are sort of an anti-pattern in any programming language, though I must say that breaking the rules isn't always a bad choice.